b2sdk/_internal/scan/folder.py

```diff
@@ -219,91 +219,59 @@ def _walk_relative_paths(
-                continue
-
-            if policies_manager.exclude_all_symlinks and local_path.is_symlink():
-                if reporter is not None:
-                    reporter.symlink_skipped(str(local_path))
-                continue
-
             if local_path.is_dir():
-                name += '/'
                 if policies_manager.should_exclude_local_directory(str(relative_file_path)):
-                    continue
-
-            # remove the leading './' from the relative path to ensure backward compatibility
-            relative_file_path_str = str(relative_file_path)
-            if relative_file_path_str.startswith("./"):
-                relative_file_path_str = relative_file_path_str[2:]
-            names.append((name, local_path, relative_file_path_str))
-
-        # Yield all of the answers.
-        #
-        # Sorting the list of triples puts them in the right order because 'name',
-        # the sort key, is the first thing in the triple.
-        for (name, local_path, relative_file_path) in sorted(names):
-            if name.endswith('/'):
+                    continue  # Skip excluded directories
+                # Recurse into directories
                 yield from self._walk_relative_paths(
-                    local_path,
-                    relative_file_path,
-                    reporter,
-                    policies_manager,
-                    visited_symlinks,
+                    local_path, relative_file_path, reporter, policies_manager, visited_symlinks
                 )
             else:
-                # Check that the file still exists and is accessible, since it can take a long time
-                # to iterate through large folders
-                if is_file_readable(str(local_path), reporter):
+                try:
                     file_mod_time = get_file_mtime(str(local_path))
                     file_size = local_path.stat().st_size
+                except OSError:
+                    file_mod_time, file_size = -1, -1
+                local_scan_path = LocalPath(
+                    absolute_path=self.make_full_path(str(relative_file_path)),
+                    relative_path=str(relative_file_path),
+                    mod_time=file_mod_time,
+                    size=file_size
+                )
+                if policies_manager.should_exclude_local_path(local_scan_path):
+                    continue  # Skip excluded files
 
-                    local_scan_path = LocalPath(
-                        absolute_path=self.make_full_path(str(relative_file_path)),
-                        relative_path=str(relative_file_path),
-                        mod_time=file_mod_time,
-                        size=file_size,
-                    )
-
-                    if policies_manager.should_exclude_local_path(local_scan_path):
-                        continue
-
+                if is_file_readable(str(local_path), reporter):
```

Comment:
`is_file_readable` doesn't just repeat half of the work made already? i.e. `os.path.exists` call can be easly replaced by the .stat() we do above right?

the other half is `os.access`

also since `is_file_readable` is not part of public API if I see correctly, we can just completely remove it as it was used only in this place and use exception value from the above.

For safety I would recommend still calling `os.access` since I don't know if you can guarantee that file is readable just because you see its .stat().